homeassistant/components/google_assistant/__init__.py

```diff
@@ -36,17 +47,58 @@
             vol.Required(CONF_ACCESS_TOKEN): cv.string,
             vol.Optional(CONF_EXPOSE_BY_DEFAULT): cv.boolean,
             vol.Optional(CONF_EXPOSED_DOMAINS): cv.ensure_list,
+            vol.Optional(CONF_AGENT_USER_ID,
+                         default=DEFAULT_AGENT_USER_ID): cv.string,
+            vol.Optional(CONF_HOMEGRAPH_API_KEY): cv.string
         }
     },
     extra=vol.ALLOW_EXTRA)
 
 
+@bind_hass
+def request_sync(hass):
+    """Request sync."""
+    hass.services.call(DOMAIN, SERVICE_REQUEST_SYNC)
+
+
 @asyncio.coroutine
 def async_setup(hass: HomeAssistant, yaml_config: Dict[str, Any]):
     """Activate Google Actions component."""
     config = yaml_config.get(DOMAIN, {})
-
+    agent_user_id = config.get(CONF_AGENT_USER_ID)
+    api_key = config.get(CONF_HOMEGRAPH_API_KEY)
+    descriptions = yield from hass.async_add_job(
```

Comment:
This is reading a file from disk. We should only do this if we are going to use this (when api_key is not None)